internal/lsp/server.go

```diff
@@ -517,7 +518,7 @@ func (l *LanguageServer) handleTextDocumentHover(
 	if !ok {
 		l.logError(fmt.Errorf("could not get builtins for uri %q", params.TextDocument.URI))
 
-		return struct{}{}, nil
+		return nil, nil
```

Comment:
I'd also add a link to https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_hover just to be really clear that this is an important return value.

Reply:
I added comments where we return null as per the specification